ModularTegustation/tegu_mobs/lc13_outskirtdwellers.dm

```diff
@@ -416,25 +416,47 @@
 	butcher_difficulty = 3
 	butcher_results = list(/obj/item/ksyringe = 1, /obj/item/assembly/flash/handheld = 1)
 	deathmessage = "buzzes as he falls out of the air."
-	density = TRUE
+	density = FALSE
 	search_objects = 1
-	var/current_size = RESIZE_DEFAULT_SIZE
 	del_on_death = TRUE
 
-/mob/living/simple_animal/hostile/kcorp/drone/AttackingTarget()
-	. = ..()
-	if(ishuman(target))
-		var/mob/living/carbon/human/L = target
-		icon_state = "kcorp_drone_angry"
-		icon_living = "kcorp_drone_angry"
-		animate(src, time = 1, loop = 0)
-		playsound(src, 'sound/weapons/flash.ogg', 100, TRUE)
-		L.Paralyze(3 SECONDS)
+	var/can_act = TRUE //If flashing, turns FALSE so we don't attack/move
+	var/flash_cooldown
+	var/flash_cooldown_time = 10 SECONDS
+	var/flash_range = 5 //Range that the flash hits
+
+/mob/living/simple_animal/hostile/kcorp/drone/proc/Flash()
+	icon_state = "kcorp_drone_angry"
+	can_act = FALSE
+	SLEEP_CHECK_DEATH(2 SECONDS) //Enough to run away, but not easily
+	playsound(src, 'sound/weapons/flash.ogg', 100, TRUE)
+
+	for (var/mob/living/L in viewers(src, null)) //The actual flashing
```

Comment:
That's uh, not how you use that proc? `viewers(Depth=world.view,Center=usr)`

Reply:
Fixed